server/src/main/java/io/crate/expression/scalar/arithmetic/RoundFunction.java

```diff
@@ -62,6 +69,36 @@ public static void register(Functions.Builder module) {
                     }
                 }
             );
+
+            module.add(
+                Signature.builder(NAME, FunctionType.SCALAR)
+                    .argumentTypes(DataTypes.DOUBLE.getTypeSignature(),
+                        DataTypes.INTEGER.getTypeSignature())
+                    .returnType(DataTypes.NUMERIC.getTypeSignature())
+                    .features(Scalar.Feature.DETERMINISTIC, Scalar.Feature.STRICTNULL)
+                    .build(),
+                RoundFunction::roundWithPrecision
+            );
         }
     }
+
+    private static Scalar<Number, Number> roundWithPrecision(Signature signature, BoundSignature boundSignature) {
+        return new Scalar<>(signature, boundSignature) {
+
+            @Override
+            public Number evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Number>... args) {
+                Number n = args[0].value();
+                Number nd = args[1].value();
+                if (n == null || nd == null) {
+                    return null;
+                }
+                double val = n.doubleValue();
+                int numDecimals = nd.intValue();
+                if(numDecimals < 0) {
```

Comment:
This is needed to keep the correct scale for `round(123.123,-1)` i.e. `120` instead of `1.2E+2`